Server standalone lambda disconnect

elpaso opened the pull request:
This fixes a couple of issues with the development server under load.

There are still too many processEvents in server code that might call the connection destructor while inside the lambda, but with parallel rendering disabled the development server behaves much better.

elpaso committed:
Disconnect lambda on connectio close

elpaso committed:
Disable parallel rendering

On load, parallel rendering internal loop calls
the readyRead slot on a deleted connection.

elpaso commented:
@pblottiere I could use your help here :) 
Maybe you find a way to make it work with parallel rendering on.

elpaso committed:
Typo

pblottiere commented:
Hi @elpaso,

> Maybe you find a way to make it work with parallel rendering on.

I'll give it  try, but feel free to merge this PR in the meantime :+1:

elpaso committed:
Server standalone: circumvent evil event loops

QGIS Server has calls to processEvents and internal
event loops that can change connection state calling
disconnect() and triggering a delete of the connection
while still in the readyRead lambda.

By disconnecting the signals and reconnecting them
after the QGIS server process has finished we can
still safely delete the connection taking its
state into account.

elpaso commented:
@pblottiere I think I've found a way around the event loop and processEvents hell.
